data/text_sampling/extend_tabular.py

```diff
@@ -0,0 +1,254 @@
+import glob
+import multiprocessing as mp
+import os
+import random
+import time
+from functools import partial
+
+import deepsmiles
+import pandas as pd
+import pubchempy as pcp
+import requests
+import selfies
+from rdkit import Chem
+
+# tucan needs very likely python 3.10
+# from tucan.canonicalization import canonicalize_molecule
+# from tucan.io import graph_from_molfile_text
+# from tucan.serialization import serialize_molecule
+from utils import load_yaml
+
+
+def augment_smiles(smiles: str, int_aug: int = 50, deduplicate: bool = True) -> str:
+    """
+    Takes a SMILES (not necessarily canonical) and returns `int_aug` random variations of this SMILES.
+    """
+
+    mol = Chem.MolFromSmiles(smiles)
```

Comment:
the question would also be if we play with and without explicit hydrogen (for a separate PR)

Reply:
we could do that, I guess this will be just another (kw)arg?